src/probability/martingale.lean

```diff
@@ -243,6 +244,38 @@ hf.sub_supermartingale hg.supermartingale
 
 end submartingale
 
+section
+
+lemma submartingale_of_set_integral_le [is_finite_measure μ]
+  {f : ι → α → ℝ} (hadp : adapted ℱ f) (hint : ∀ i, integrable (f i) μ)
+  (hf : ∀ i j : ι, i ≤ j → ∀ s : set α, measurable_set[ℱ i] s →
+    ∫ x in s, f i x ∂μ ≤ ∫ x in s, f j x ∂μ) :
+  submartingale f ℱ μ :=
+begin
+  refine ⟨hadp, λ i j hij, _, hint⟩,
+  -- need to show `f i ≤ᵐ[μ] μ[f j|_]`. This is equivalent to `f i ≤ᵐ[μ.trim _] μ[f j|_]`
+  -- since both sides are `ℱ i`-measurable
+  suffices : f i ≤ᵐ[μ.trim (ℱ.le i)] μ[f j| ℱ.le i],
```

Comment:
Could you add a lemma `ae_le_of_ae_le_trim` (similar to `ae_eq_of_ae_eq_trim`) which proves the implication you need for any two functions? And perhaps also a new lemma `ae_le_trim_iff`, similar to `ae_eq_trim_iff`?

This new lemma can be useful:
```lean
lemma ae_of_ae_trim {α : Type*} {m m0 : measurable_space α} (hm : m ≤ m0) {μ : measure α}
  {P : α → Prop} (h : ∀ᵐ x ∂(μ.trim hm), P x) :
  ∀ᵐ x ∂μ, P x :=
measure_eq_zero_of_trim_eq_zero hm h
```